bob-common/src/metrics/exporters/global_exporter/rate_processor.rs

```diff
@@ -0,0 +1,65 @@
+const BUFFER_SIZE: usize = 10_485_760; // 10 Mb
+use metrics::Key;
+use std::time::Duration;
+use std::{collections::HashMap, time::Instant};
+use tokio::sync::mpsc::{channel, Receiver, Sender};
+
+const RATE_INTERVAL: Duration = Duration::from_secs(5);
+
+type TimeStamp = Instant;
+
+pub(super) struct RateProcessor {
+    tx: Sender<(String, u64, TimeStamp)>,
+}
+
+impl RateProcessor {
+    pub(super) fn run_task() -> Self {
+        let (tx, rx) = channel(BUFFER_SIZE);
+        tokio::spawn(rate_processor_task(rx));
+        Self { tx }
+    }
+
+    pub(super) fn process(&self, key: &Key, value: u64) {
+        if let Err(e) = self
+            .tx
+            .try_send((key.name().to_owned(), value, Instant::now()))
+        {
+            error!("Can't send counter to task with rates (reason: {})", e);
```

Comment:
You can process TrySendError:
Full - try again or just drop
Closed - shutdown whole processor, receiver closed and won't be restored

Reply:
I'll do that, but this method is called from immutable method of global exporter (which is defined in trait from `metrics`) and if we want to do some shutdown logic on closed, we need:
1. Store rate processor in global exporter with some interior mutability wrapper (and this wrapper should be thread-safe, because I've already have had problems with `metrics` crate: their unsafe code makes possible `increment_counter` sync function to run concurrently)
2. Store state of processor via interior mutability pattern (also thread-safe implementation)
3. Unsafe code (and sync logic also, which should be with doubled attention)

Current version (for me) seems better: if error is `Closed`, then receiver was dropped due to rate_processor_task function end (we don't need to cleanup it) and `try_send` will just return error

I'll implement it with `is_active` atomic bool marker in processor